src/Illuminate/Database/Console/Migrations/RollbackCommand.php

```diff
@@ -85,6 +95,34 @@ protected function getOptions()
             ['pretend', null, InputOption::VALUE_NONE, 'Dump the SQL queries that would be run'],
             ['step', null, InputOption::VALUE_OPTIONAL, 'The number of migrations to be reverted'],
             ['batch', null, InputOption::VALUE_REQUIRED, 'The batch of migrations (identified by their batch number) to be reverted'],
+            ['select', null, InputOption::VALUE_NONE, 'Select the migrations to rollback'],
         ];
     }
+
+    /**
+     * Get the list of migrations selected by the user for rollback.
+     *
+     * This method presents an interactive prompt allowing users to select
+     * which migrations they want to rollback from the existing migrations
+     * in the database.
+     *
+     * @return array<int, object> Returns an array of migration records, or empty array if no migrations exist
+     */
+
+    private function getMigrationsForRollbacks()
+    {
+        $migrationsInstance = DB::table('migrations');
+
+        if ($migrationsInstance->count() > 0) {
+            $options = multiselect(
+                label: 'Which migrations would you like to rollback (leave blank to default behaviour)',
+                options: $migrationsInstance->pluck('migration')->toArray(),
+                hint: 'Use the space bar to select options.',
+                scroll: 10,
+                required: true,
+            );
+            return $migrationsInstance->whereIn('migration', $options)->get()->toArray();
+        }
+        return [];
```

Comment:
Not sure if you want to swap this:
```suggestion
        if ($migrationsInstance->count() === 0) {
            return [];
        }

        $options = multiselect(
            label: 'Which migrations would you like to rollback (leave blank to default behaviour)',
            options: $migrationsInstance->pluck('migration')->toArray(),
            hint: 'Use the space bar to select options.',
            scroll: 10,
            required: true,
        );
        return $migrationsInstance->whereIn('migration', $options)->get()->toArray();
```